### PR TITLE
Automated test as workflow

### PR DESCRIPTION
**Pull request by mml5bg**
Changes needed to introduce automated tests for devcontainer environment as a workflow.

### CODE REVIEW

File: src/python-app/workflows/check-devcontainer.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Build and run dev container task
+        uses: devcontainers/ci@v0.3
+        with:
+          runCmd: ./tests/automated_tests/test.sh
```
Comment (MP91) [inline src/python-app/workflows/check-devcontainer.yml:42]: why did you remove the push?
```suggestion
        with:
          runCmd: ./tests/automated_tests/test.sh
          push: never
```

File: src/python-app/workflows/check-devcontainer.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  workflow_dispatch:
+  push:
+    # Run only on branches/commits and not tags
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  schedule:
+    - cron: "1 1 * * 1"
```
Comment (MP91) [inline src/python-app/workflows/check-devcontainer.yml:28]: isn't that to often? running this on PR should be enough

**Review by MP91 — CHANGES_REQUESTED** [review]

File: src/python-app/workflows/check-devcontainer.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# This file is maintained by velocitas CLI, do not modify manually. Change settings in .velocitas.json
```
Comment (doosuu) [inline src/python-app/workflows/check-devcontainer.yml:1]: Remove this line, it is added by the CLI when syncing.

File: src/python-app/workflows/check-devcontainer.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+# Copyright (c) 2022 Robert Bosch GmbH and Microsoft Corporation
```
Comment (doosuu) [inline src/python-app/workflows/check-devcontainer.yml:2]:
```suggestion
# Copyright (c) 2023 Robert Bosch GmbH
```

File: src/python-app/workflows/check-devcontainer.yml (hunk 1 above)
Comment (doosuu) [inline src/python-app/workflows/check-devcontainer.yml:42]: Where are the scripts in this PR?
Why chose bash over Python as scripting language? I'd prefer to use Python and Pytest also for reports

**Review by doosuu — CHANGES_REQUESTED** [review]

File: src/common/workflows/check-devcontainer.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  workflow_dispatch:
+  push:
+    # Run only on branches/commits and not tags
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  schedule:
+    - cron: "1 1 * * 1"
```
Comment (MP91) [inline src/common/workflows/check-devcontainer.yml:27]: why do we need this schedule? what should change that is not controlled by us?

Comment (MP91) [inline src/common/workflows/check-devcontainer.yml:27]: please change this to run daily/nightly

**Review by MP91 — CHANGES_REQUESTED** [review]

**Review by MP91 — APPROVED** [review]
lgtm

**Review by doosuu — APPROVED** [review]
lgtm

**Pull request merged**